packages/api/app/Controllers/Http/TechnologyCostController.js

```diff
@@ -0,0 +1,145 @@
+const Technology = use('App/Models/Technology');
+const TechnologyCost = use('App/Models/TechnologyCost');
+const Cost = use('App/Models/Cost');
+
+const { antl, errors, errorPayload, getTransaction } = require('../../Utils');
+
+const getFields = (request) =>
+	request.only([
+		'funding_required',
+		'funding_type',
+		'funding_value',
+		'funding_status',
+		'notes',
+		'costs',
+	]);
+
+class TechnologyCostController {
+	/**
+	 * Show a list of all technology costs.
+	 * GET technology_costs?technologyId=
```

Comment:
Estava vendo aqui e para manter o padrão acho que essa rota deveria ser GET `technology/:id/costs` o que acha?

Reply:
Nesse caso @nicholasio é melhor deixar do jeito que está pelo seguinte motivo:
No `HandleParams.js` ele obtem o recurso que vai ser usado através da url:
```javascript
const get_resource = request.url().split('/')[1];
const resource = get_resource;
```
Nesse caso ele pegaria `technology` o que não faria sentido, por isso eu coloquei GET `technology_costs` porque essa rota lista os technologies costs, todos (se não for passado o technologyId) ou apenas o da tecnologia se for passado o technologyId.
Outra coisa que não daria certo nessa rota que você mencionou seria o fato dele listar apenas tecnology cost da tecnologia passada e não todos que seria o propósito da rota index.

Reply:
Outra alternativa para corrigir esse "problema" seria o middleware `HandleParams.js` receber um parâmetro HandleParams:{resource}, e:
```javascript
const get_resource = resource || request.url().split('/')[1];
const resource = get_resource;
```

Reply:
Eu acho que não precisamos de uma rota que lista custos de tecnologias diferente, acho que vale a pena implementar essa solução passando o resource para o middleware.